script.js

```diff
@@ -219,76 +239,37 @@ const createSelectOptions = (episode) => {
   number = number < 10 ? "0" + number : number;
   return `S${season}E${number}`;
 }
 
-// const allShows = getAllShows()
-const showSelect = document.getElementById("shows")
-
-// allShows.forEach(show=>{
-//   const showOption = document.createElement("option")
-  
-//   showOption.innerText = show.name
-//   showOption.id = show.id
-//   show.value = show.id
-//   showSelect.appendChild(showOption)
-// })
+const episodeContainerEvent = (e) => {
+  //console.log(e.currentTarget);
+  let ID = e.currentTarget.id;
+  console.log(e.currentTarget);
 
-// showSelect.onchange = function() {
-//   const showID = document.getElementById("shows").value
-//   console.log(showID)
-//   rootElem.innerHTML = " ";
-//   searchField.innerHTML = " "
-//   allShows.forEach(show=>{
-//     if(showID==show.name){
-//       console.log(show.id)
-//       url = `https://api.tvmaze.com/shows/${show.id}/episodes`;
-//       console.log(url)
+  if (allEpisodes[0].url.includes(".com/shows")) {
+    url = `https://api.tvmaze.com/shows/${ID}/episodes`;
+    //const showName = document.querySelector(".show-name");
+    //showName.innerText = `Your chosen show: ${text}`;
+    rootElem.innerHTML = "";
+    setup();
+    dropdown.innerHTML = "";
+  } else if (allEpisodes[0].url.includes(".com/episodes")) {
+    const pickedEpisodes = allEpisodes.filter((episode) => {
+      return episode.id == ID;
       
-//     }
-    
-//     setup()
-//     makePageForEpisodes(allEpisodes)
-//   })
-// }
-// console.log(setup())
-//console.log(allShows);
-window.onload = setup;
-
-
-
-// let allEpisodes = getAllEpisodes();
-// let rootElem = document.getElementById("root");
-// let episodeTab, episodeName, episodeThumbnail, episodeSummary, searchField, allEpisodesDiv;
-
-// searchField = rootElem.appendChild(document.createElement("input"));
-// searchField.id = "searchField"
-// allEpisodesDiv = rootElem.appendChild(document.createElement("div"));
-// allEpisodesDiv.className = "allEpisodesDiv"
-
-// searchField.addEventListener("keyup", ()=>{
-//   // allEpisodes.filter()
-
-// })
+    });
+    rootElem.innerHTML = "";
+    makePageForEpisodes(pickedEpisodes);  
+};
+createButton();
+};
 
-//  allEpisodes.forEach(item=>{
-//  episodeTab = allEpisodesDiv.appendChild(document.createElement("section"));
-//  episodeName = episodeTab.appendChild(document.createElement("h1"));
-//  episodeThumbnail = episodeTab.appendChild(document.createElement("img"));
-//  episodeSummary = episodeTab.appendChild(document.createElement("p"))
-//  episodeTab.className = "episodeTab";
-//  episodeName.className = "episodeName";
-//  episodeThumbnail.className = "episodeThumbnail";
-//  episodeSummary.className = "episodeSummary";
-//  episodeName.innerText = `${item.name} - S${('0' + item.season).slice(-2)}E${('0' + item.number).slice(-2)}` ;
-//  episodeThumbnail.src = item.image.medium;
-//  episodeSummary.innerHTML = item.summary
-// })
+window.onload = setup;
 
```

Comment:
After reviewing my code again after awhile I think that I could do some improvements to it. First of all I should make more use of comments to describe the different functions in my code. I should delete the commented out pieces of old code to make it more readable as well, I have left in those on purpose to remind me of the process of creating the website. Having another look at my deployed website I can say that I am proud of my work mainly regarding the functionality, everything works as it supposed to and there are no annoying glitches when searching back and forth. The presentation of the website could be better, but this wasn't my main concern regarding this project. Also I have to thank my colleagues an my tech buddy, as parts of the project were collaborative and I have learned a lot about the importance of that in any software development project.